src/main/java/nextstep/subway/section/domain/SortedSection.java

```diff
@@ -0,0 +1,71 @@
+package nextstep.subway.section.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.Function;
+import nextstep.subway.station.domain.Station;
+
+import static java.util.stream.Collectors.toMap;
+
+public class SortedSection {
+
+    private static final String MESSAGE_NOT_FOUND_UPSTATION = "상행역을 찾을 수 없습니다.";
+
+    private final List<Section> sections;
+
+    public SortedSection(LineSections lineSections) {
+
+        List<Section> sortedSections = lineSections.getSections();
+
+        Section startSection = getStartSection(sortedSections);
+        this.sections = Collections.unmodifiableList(linkToLastSection(startSection, sortedSections));
+    }
+
+    private Section getStartSection(List<Section> sections) {
+
+        Map<Station, Station> stationMap =
+            sections.stream()
+                    .collect(toMap(Section::getDownStation,
+                                   Section::getUpStation));
+
+        Entry<Station, Station> startEntry =
+            stationMap.entrySet()
+                      .stream()
+                      .filter(entry -> !stationMap.containsKey(entry.getValue()))
+                      .findAny()
+                      .orElseThrow(
+                          () -> new IllegalStateException(MESSAGE_NOT_FOUND_UPSTATION));
+
+        return sections.stream()
+                       .filter(section -> section.equalsUpStation(startEntry.getValue()))
+                       .filter(section -> section.equalsDownStation(startEntry.getKey()))
+                       .findAny()
+                       .orElseThrow(() -> new IllegalStateException(MESSAGE_NOT_FOUND_UPSTATION));
+    }
```

Comment:
```suggestion
                Map<Station, Section> stationMap =
            sections.stream()
                    .collect(toMap(Section::getDownStation,
                                   Function.identity()));

        Entry<Station, Section> startEntry =
            stationMap.entrySet()
                      .stream()
                      .filter(entry -> !stationMap.containsKey(entry.getValue().getUpStation()))
                      .findAny()
                      .orElseThrow(
                          () -> new IllegalStateException(MESSAGE_NOT_FOUND_UPSTATION));

        return startEntry.getValue();
    }
```

map 의 value를 section으로 저장하면, 
key / value 와 일치하는 section을 찾기 위해 다시 탐색할 필요가 없지 않을까요 ?

Reply:
리팩토링하다가 애매하게 옮겨 버렸네요...
원래 `SortedStation` 에 들어있던 로직인데 옮기면서 하나만 바꿔버렸어요 ㅋㅋ...
정확한 피드백 감사드립니다~